lighthouse-core/scripts/internal-analysis/download-issues.js

```diff
@@ -0,0 +1,136 @@
+ * @property {Array<{event: 'labeled'|'unlabeled'|'closed'|'assigned', actor: {login: string}, assignee?: {login: string}, label?: {name: string}, created_at: string}>} events
+ */
+
+/** @param {number} ms @return {Promise<void>} */
+function wait(ms) {
+  return new Promise(r => setTimeout(r, ms));
+}
+
+/**
+ * @param {AugmentedGitHubIssue} issue
+ */
+async function fetchAndInjectEvents(issue) {
+  process.stdout.write(`Fetching ${issue.events_url}...\n`);
+  const response = await fetch(issue.events_url, {headers: HEADERS});
+  const events = await response.json();
+  issue.events = events;
+}
+
+/**
+ * @param {AugmentedGitHubIssue} issue
+ */
+async function fetchAndInjectComments(issue) {
+  process.stdout.write(`Fetching ${issue.comments_url}...\n`);
+  const response = await fetch(issue.comments_url, {headers: HEADERS});
+  if (!response.ok) throw new Error(`Invalid API response: ${await response.text()}`);
+  const comments = await response.json();
+  issue.comments = comments;
+  if (!Array.isArray(issue.comments)) {
+    console.warn('Comments was not an array', issue.comments); // eslint-disable-line no-console
+    issue.comments = [];
+  }
+
+  if (issue.pull_request) {
+    const prCommentsUrl = issue.comments_url
+      .replace('/issues/', '/pulls/')
+      .replace('/comments', '/reviews');
+    process.stdout.write(`Fetching ${prCommentsUrl}...\n`);
+    const response = await fetch(prCommentsUrl, {headers: HEADERS});
+    if (!response.ok) throw new Error(`Invalid API response: ${await response.text()}`);
+    const comments = await response.json();
+    issue.comments = issue.comments.concat(comments);
+  }
+}
+
+/**
+ * @param {string} [urlToStartAt]
+ * @return {Promise<Array<AugmentedGitHubIssue>>}
+ */
+async function downloadIssues(urlToStartAt) {
+  const url = new URL(`/repos/GoogleChrome/lighthouse/issues`, GITHUB_API);
+  url.searchParams.set('state', 'all');
+  url.searchParams.set('since', START_FROM.toISOString());
+  const urlToFetch = urlToStartAt || url.href;
+  process.stdout.write(`Fetching ${urlToFetch}...\n`);
+  const response = await fetch(urlToFetch, {headers: HEADERS});
+  if (!response.ok) throw new Error(`Invalid API response: ${await response.text()}`);
+  /** @type {Array<AugmentedGitHubIssue>} */
+  const issues = await response.json();
+  const linkHeader = response.headers.get('link') || '';
+  const nextLink = linkHeader
+    .split(',')
+    .find(link => link.split(';')[1].includes('rel="next"'));
+  const nextUrlMatch = (nextLink && nextLink.match(/<(https.*?)>/)) || [];
+  const nextUrl = nextUrlMatch[1] || '';
+  const restOfIssues = nextUrl ? await downloadIssues(nextUrl) : [];
+
+  // Yes really do this in series to avoid hitting abuse limits of GitHub API
+  for (const issue of issues) {
+    await Promise.all([
+      fetchAndInjectEvents(issue).catch(async err => {
+        console.error('Events failed! Trying again', err); // eslint-disable-line no-console
```

Comment:
I had random github API timeouts about once or twice a run, and since I didn't want to implement "pick back up where you left off" functionality it was a giant pain in the butt if 1/1000 requests failed.

eventually I think we can do resumption and this is less of a big deal